game-core/src/main/java/games/strategy/triplea/ui/MovePanel.java

```diff
@@ -555,7 +556,8 @@ private void selectEndPoint(final Territory territory) {
             units.addAll(
                 getUnitsToUnload(
                     route, CollectionUtils.getMatches(selectedUnits, getUnloadableMatch())));
-            units.addAll(CollectionUtils.getMatches(selectedUnits, getUnloadableMatch().negate()));
+            units.addAll(
+                CollectionUtils.getMatches(unitsThatCanMoveOnRoute, getUnloadableMatch().negate()));
```

Comment:
The 'negate' is confusing. A comment to clear it up or adding a new matcher so the condition is not negated would help. It's hard to understand why we are taking the units that cannot move on the route and adding that to the `units` collection.

Reply:
I think you are misunderstanding. We are taking the non-unloadable units (essentially air/sea units) and adding them in.

Reply:
I know I am misunderstanding :)
That is the problem, this is a very confusing thing to read! It seems like the code is doing the opposite with the 'negate'. With the selected units we seems like we were grabbing all units in the territory as a tautology. It's not clear why this addition is needed and what's it's really doing. I don't think future code maintainers would have better luck.

Reply:
A comment goes a long way.....